node-cli/watch.js

```diff
@@ -0,0 +1,28 @@
+const output = require("./output");
+const index = require("./index");
+
+function listener(tasks, changes) {
+  changes.deletions.forEach((index) => {
+    // Deleted objects cannot be accessed directly,
+    // but we can update a UI list, etc. knowing the index.
+    output.watchResult("Task Deleted", "A task has been deleted.");
+  });
+
+  changes.insertions.forEach((index) => {
+    let insertedTask = tasks[index];
+    output.watchResult("Task Created", JSON.stringify(insertedTask, " ", 3));
+  });
+
+  changes.modifications.forEach((index) => {
+    let modifiedTask = tasks[index];
+    output.watchResult("Task Modified", JSON.stringify(modifiedTask, " ", 3));
+  });
+}
+
+async function watchForChanges() {
+  const realm = await index.getRealm();
+  const tasks = await realm.objects("Task");
```

Comment:
There shouldn't be a need to await the returned value here.

Reply:
@kraenhansen you have to await the getRealm or else it fails.

Reply:
```suggestion
  const tasks = realm.objects("Task");
```

Sure, but realm.objects doesn't return a promise.